### PR TITLE
Added scale-to-zoom gesture support to the Gallery grid demo

### PR DESCRIPTION
**Pull request by HansMuller**
Added basic support for the scale input gesture to the Gallery grid demo.

Added a flingVelocity parameter to the scaleEnd() callback.

### CODE REVIEW

File: examples/flutter_gallery/lib/demo/grid_list_demo.dart [hunk 1]
```diff
@@ -24,7 +28,112 @@ class Photo {
...
+    );
+     return new Point(point.x.clamp(0.0, bottomRight.x), point.y.clamp(0.0, bottomRight.y));
+  }
+
+  void _handleFlingAnimation() {
+    setState(() {
+      _focalPoint = _flingAnimation.value;
+      _lastFocalPoint = _focalPoint;
+    });
+
```
Comment (abarth) [inline examples/flutter_gallery/lib/demo/grid_list_demo.dart]: extra blank line.

File: packages/flutter/lib/src/gestures/scale.dart [hunk 2]
```diff
@@ -35,7 +36,13 @@ typedef void GestureScaleStartCallback(Point focalPoint);
 typedef void GestureScaleUpdateCallback(double scale, Point focalPoint);
 
 /// Signature for when the pointers are no longer in contact with the screen.
-typedef void GestureScaleEndCallback();
+typedef void GestureScaleEndCallback(Velocity flingVelocity);
```
Comment (abarth) [inline packages/flutter/lib/src/gestures/scale.dart]: I wonder if these should take details objects.  That's the pattern we've adopted for the other gestures.

**Review by abarth — APPROVED** [review]
LGTM

File: packages/flutter/lib/src/gestures/scale.dart (hunk 2 above)
Comment (HansMuller) [inline packages/flutter/lib/src/gestures/scale.dart]: I agree. I can take a stab at that in a separate PR.

**Pull request merged**